subprojects/model-core/src/integTest/groovy/org/gradle/api/provider/PropertyIntegrationTest.groovy

```diff
@@ -888,4 +888,22 @@ project.extensions.create("some", SomeExtension)
         where:
         fieldModifier << ["", "final"]
     }
+
+    def "filter is evaluated lazily"() {
```

Comment:
🧪 We should also test a `filter` chain being used as an `@Input` for some tasks:
* when the filter chain is carrying task dependencies
* when the filter is applied to a provider with no value set